### PR TITLE
mixin: Add block deletion panels for compactor dashboard

### PR DESCRIPTION
**Pull request by Sayrus**
* [x] I added CHANGELOG entry for this change.
* [X] Change is not relevant to the end user.

## Changes

This PR adds a new row to the Thanos Compact Dashboard. This row contains:
- The rate at which blocks are deleted from S3: This is useful as it avoids reading logs or manually querying the metric. In 0.15, the deletion is done after downsampling so it allows us to monitor whether it is actually called or not. In 0.16, this process is done concurrently thanks to #3115 . This graph then allows us to monitor if the async context is doing its job.
- The error rate when trying to delete from S3
- The rate at which blocks are marked for deletion, not only from GC but also from retention policy

## Verification

Our compactor isn't deleting anything at the moment so the graphs are empty.

**Notes**
Metrics will need to be renamed following #3309 as they use the `thanos_compact_*` format.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ We use *breaking* word for marking changes that are not backward compatible (rel
 - [#2521](https://github.com/thanos-io/thanos/pull/2521) Sidecar: add `thanos_sidecar_reloader_reloads_failed_total`, `thanos_sidecar_reloader_reloads_total`, `thanos_sidecar_reloader_watch_errors_total`, `thanos_sidecar_reloader_watch_events_total` and `thanos_sidecar_reloader_watches` metrics.
 - [#2412](https://github.com/thanos-io/thanos/pull/2412) ui: add React UI from Prometheus upstream. Currently only accessible from Query component as only `/graph` endpoint is migrated.
 - [#2532](https://github.com/thanos-io/thanos/pull/2532) Store: Added hidden option for experimental caching bucket, that can cache chunks into shared memcached. This can speed up querying and reduce number of requests to object storage.
+- [#3380](https://github.com/thanos-io/thanos/pull/3380) compact: add deletion metrics to mixins.
```
Comment (kakkoyun) [inline CHANGELOG.md:27]:
```suggestion
- [#3380](https://github.com/thanos-io/thanos/pull/3380) mixin: add deletion metrics for compactor.
```

**Review by kakkoyun — COMMENTED** [review]
Thanks for the PR. I can see the use case.
Let's wait for the renaming PR to be merged.

**Comment by metalmatze** [comment]
Renaming PRs should be merged.

**Comment by Sayrus** [comment]
Updated for new names :ok_hand:

**Review by 330981 — APPROVED** [review]

**Comment by kakkoyun** [comment]
@Sayrus Looks good so far. You need to run `make examples` and commit changes to make CI happy.

File: CHANGELOG.md [hunk 2]
```diff
@@ -209,6 +209,7 @@ sse_config:
 - [#2579](https://github.com/thanos-io/thanos/pull/2579) Store: Experimental caching bucket can now cache metadata as well. Config has changed from #2532.
 - [#2526](https://github.com/thanos-io/thanos/pull/2526) Compact: In case there are no labels left after deduplication via `--deduplication.replica-label`, assign first `replica-label` with value `deduped`.
 - [#2621](https://github.com/thanos-io/thanos/pull/2621) Receive: Added flag to configure forward request timeout. Receive write will complete request as soon as quorum of writes succeeds.
+- [#3380](https://github.com/thanos-io/thanos/pull/3380) mixin: add deletion metrics for compactor.
```
Comment (kakkoyun) [inline CHANGELOG.md:212]:
```suggestion
- [#3380](https://github.com/thanos-io/thanos/pull/3380) mixin: Add block deletion panels for compactor dashboards.
```

**Review by kakkoyun — COMMENTED** [review]
nit: Let's be super clear on what we actually add in the changelog.

Then we can merge.

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**